src/rsdiv/evaluation/ranking_metrics.py

```diff
@@ -0,0 +1,56 @@
+from typing import Callable, Hashable, Sequence, TypeVar
+
+import numpy as np
+
+ItemNameType = TypeVar("ItemNameType", bound=Hashable)
```

Comment:
The practice is to use a single capital letter for `User-defined generic types`:
https://docs.python.org/3/library/typing.html#user-defined-generic-types